hs_tracking/management/commands/stats.py

```diff
@@ -0,0 +1,222 @@
+
+    def monthly_orgs_counts(self, start_date, end_date):
+        profiles = UserProfile.objects.filter(user__date_joined__lte=end_date)
+        org_count = profiles.values('organization').distinct().count()
+        self.print_var("monthly_orgs_counts", org_count, (start_date, end_date))
+
+    def monthly_users_by_type(self, start_date, end_date):
+        user_types = UserProfile.objects.values('user_type').distinct()
+        for ut in [_['user_type'] for _ in user_types]:
+            ut_users = User.objects.filter(userprofile__user_type=ut)
+            sessions = hs_tracking.Session.objects.filter(
+                Q(begin__gte=start_date) &
+                Q(begin__lte=end_date) &
+                Q(visitor__user__in=ut_users)
+            )
+            self.print_var("active_{}".format(ut),
+                           sessions.count(), (end_date, start_date))
+
+    def users_details(self):
+        w = csv.writer(sys.stdout)
+        fields = [
+            'first name',
+            'last name',
+            'email',
+            'user type',
+            'organization',
+            'created date',
+            'last login',
+        ]
+        w.writerow(fields)
+
+        for up in UserProfile.objects.filter(user__is_active=True):
+            last_login = up.user.last_login.strftime('%m/%d/%Y') if up.user.last_login else ""
+            values = [
+                up.user.first_name,
+                up.user.last_name,
+                up.user.email,
+                up.user_type,
+                up.organization,
+                up.user.date_joined.strftime('%m/%d/%Y'),
+                last_login,
+            ]
+            w.writerow([unicode(v).encode("utf-8") for v in values])
+
+    def resources_details(self):
+        w = csv.writer(sys.stdout)
+        fields = [
+            'title',
+            'resource type',
+            'size',
+            'federated resource file size',
+            'creation date',
+            'publication status'
+        ]
+        w.writerow(fields)
+
+        resources = BaseResource.objects.all()
+        for r in resources:
+            f_sizes = [f.resource_file.size
+                       if f.resource_file else 0
+                       for f in r.files.all()]
+            total_file_size = sum(f_sizes)
+            try:
+                f_sizes = [int(f.fed_resource_file_size)
+                           if f.fed_resource_file_size else 0
+                           for f in r.files.all()]
+                federated_resource_file_size = sum()
+            except SessionException:
+                federated_resource_file_size = "SessionException"
+            values = [
+                r.metadata.title.value,
+                r.resource_type,
+                total_file_size,
+                federated_resource_file_size,
+                r.metadata.dates.get(type="created").start_date.strftime("%m/%d/%Y"),
+                r.raccess.sharing_status,
+            ]
```

Comment:
Rearrange these columns so that the resource `creation_date` is first to be consistent with typical logging format.  This also makes the logs more human readable.